extensions/smallrye-metrics/runtime/src/main/java/io/quarkus/smallrye/metrics/runtime/SmallRyeMetricsRecorder.java

```diff
@@ -209,6 +215,94 @@ private void operatingSystemMetrics(MetricRegistry registry, List<String> names)
+        }
+    }
+
+    private void vendorOperatingSystemMetrics(MetricRegistry registry, List<String> names) {
+        OperatingSystemMXBean operatingSystemMXBean = ManagementFactory.getOperatingSystemMXBean();
+
+        // some metrics are only available in jdk internal class 'com.sun.management.OperatingSystemMXBean': cast to it.
+        // com.sun.management.OperatingSystemMXBean is not available in SubstratVM
+        if (!ImageInfo.inImageCode()) {
+            try {
+                // the cast will fail for some JVM not derived from HotSpot (J9 for example)
+                com.sun.management.OperatingSystemMXBean internalOperatingSystemMXBean = (com.sun.management.OperatingSystemMXBean) operatingSystemMXBean;
+                Metadata meta = Metadata.builder()
+                        .withName(SYSTEM_CPU_LOAD)
+                        .withType(MetricType.GAUGE)
+                        .withDisplayName("System CPU load")
+                        .withDescription("Displays the \"recent cpu usage\" for the whole system. This value is a double " +
+                                "in the [0.0,1.0] interval. A value of 0.0 means that all CPUs were idle during the recent " +
+                                "period of time observed, while a value of 1.0 means that all CPUs were actively running " +
+                                "100% of the time during the recent period being observed. " +
+                                "All values betweens 0.0 and 1.0 are possible depending of the activities going on in the " +
+                                "system. If the system recent cpu usage is not available, the method returns a negative value.")
+                        .build();
+                registry.register(meta, new LambdaGauge(() -> internalOperatingSystemMXBean.getSystemCpuLoad()));
+                names.add(SYSTEM_CPU_LOAD);
+
+                meta = Metadata.builder()
+                        .withName(PROCESS_CPU_TIME)
+                        .withType(MetricType.GAUGE)
+                        .withDisplayName("Process CPU time")
+                        .withDescription(
+                                "Displays the CPU time used by the process on which the Java virtual machine is running " +
+                                        "in nanoseconds. The returned value is of nanoseconds precision but not necessarily nanoseconds "
+                                        +
+                                        "accuracy. This method returns -1 if the the platform does not support this operation.")
+                        .build();
+                registry.register(meta, new LambdaGauge(() -> internalOperatingSystemMXBean.getProcessCpuTime()));
+                names.add(PROCESS_CPU_TIME);
+
+                meta = Metadata.builder()
+                        .withName(FREE_PHYSICAL_MEM_SIZE)
+                        .withType(MetricType.GAUGE)
+                        .withDisplayName("Process CPU time")
```

Comment:
this display name doesn't fit the metric :)

Reply:
too quick copy/paste :) => fixed